python/lsst/daf/butler/remote_butler/_remote_butler.py

```diff
@@ -272,7 +264,10 @@ def get(
         **kwargs: Any,
     ) -> Any:
         # Docstring inherited.
-        model = self._get_file_info(datasetRefOrType, dataId, collections, kwargs)
+        try:
+            model = self._get_file_info(datasetRefOrType, dataId, collections, kwargs)
+        except DatasetNotFoundError as e:
+            raise FileNotFoundError(str(e)) from e
```

Comment:
Maybe we need to talk about this oddity for the future.

Reply:
This is a good call out... I'm pretty sure this is just incorrect even though it passes tests.

I thought this was needed because of the datastore vs registry thing.  Datastore pretty consistently throws FileNotFoundError when stuff can't be found, and we're mostly acting in place of datastore here.

But I took a closer look at DirectButler and it does actually throw DatasetNotFoundError (formerly LookupError) for most common cases of not being able to lookup a ref in get().  Apparently that just isn't covered in `test_butler.py`.  

The FileNotFoundErrors being expected in the tests are actually for a couple of weird edge cases:
* changing the UUID of a ref to something invalid (line 500 in test_butler.py)
* holding on to a ref after calling `pruneDatasets(purge=True)`, then trying to call get() on it (line 417 in test_butler.py)

Arguably in both these cases it would make just as much sense to throw the DatasetNotFoundError, but because these checks are inside Datastore they happen to throw FileNotFoundError because that's what Datastore does.